moveit_ros/planning/trajectory_execution_manager/test/test_execution_manager.cpp

```diff
@@ -32,76 +32,94 @@
  *  POSSIBILITY OF SUCH DAMAGE.
  *********************************************************************/
 
-/* Author: Ioan Sucan */
+/* Author: Ioan Sucan, Cristian C. Beltran
+   Desc: Test the TrajectoryExecutionManager with MoveitCpp
+*/
 
-#include <moveit/trajectory_execution_manager/trajectory_execution_manager.h>
-#include <moveit/robot_model_loader/robot_model_loader.h>
-#include <moveit/planning_scene_monitor/planning_scene_monitor.h>
+// ROS
+#include <ros/ros.h>
 
-int main(int argc, char** argv)
-{
-  ros::init(argc, argv, "test_trajectory_execution_manager");
-
-  ros::AsyncSpinner spinner(1);
-  spinner.start();
-
-  robot_model_loader::RobotModelLoaderPtr rml(new robot_model_loader::RobotModelLoader);
-  planning_scene_monitor::PlanningSceneMonitor psm(rml);
-  trajectory_execution_manager::TrajectoryExecutionManager tem(rml->getModel(), psm.getStateMonitor(), true);
-
-  std::cout << "1:\n";
-  if (!tem.ensureActiveControllersForJoints(std::vector<std::string>(1, "basej")))
-    ROS_ERROR("Fail!");
+// Testing
+#include <gtest/gtest.h>
 
-  std::cout << "2:\n";
-  if (!tem.ensureActiveController("arms"))
-    ROS_ERROR("Fail!");
+// Main class
+#include <moveit/moveit_cpp/moveit_cpp.h>
+#include <moveit/moveit_cpp/planning_component.h>
+// Msgs
+#include <geometry_msgs/PointStamped.h>
 
-  std::cout << "3:\n";
-  if (!tem.ensureActiveControllersForJoints(std::vector<std::string>(1, "rj2")))
-    ROS_ERROR("Fail!");
+namespace moveit_cpp
+{
+class MoveItCppTest : public ::testing::Test
+{
+public:
+  void SetUp() override
+  {
+    nh_ = ros::NodeHandle("/test_execution_manager");
```

Comment:
Why do you specify a namespace here? This isn't needed, is it?
```suggestion
```

Reply:
I remove the namespace but the NodeHandle seems necessary for instantiating MoveItcpp

Reply:
Sure.